portfolio-project-matching-app/components/Nav.js

```diff
@@ -1,19 +1,42 @@
+import { useRouter } from 'next/router'
 import Link from 'next/link'
+import Button from './Button'
+import styles from './Nav.module.css'
+import { useEffect, useState } from 'react'
 // import tailwind from 'tailwind/tailwindcss.css'
 
 const Nav = () => {
+    let router = useRouter();
+    const [path, setPath] = useState('');
+    useEffect(
+        () => {
+            setPath(router.pathname);
+        }
```

Comment:
This is somewhat of a nitpick, but just wanted to point it out since it's the first place useEffect has been called. Adding the second optional parameter (the dependency array) will reduce unnecessary re-rendering. For now we could just pass an empty array, which will make it so useEffect is only ever called once:

```
    useEffect(
        () => {
            console.log('Here');
            setPath(router.pathname);
        }, []
    )
```

Reply:
Ah nice! I'm going to need to take a peek at how that all works, don't think I've used `useEffect` yet.